### ISSUE TITLE
temporary append column table insert/select error

### ISSUE THREAD

**Issue by Tao-Ma** [greenplum-db/gpdb#913]
When i use temporary append column table, I found the following error. It can be reproduce everytime, I

```
postgres=# create temp table a with (appendonly=true, orientation=column) on commit delete rows as select 1::int AS c1, 1::int AS c;
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'c1' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT 1
postgres=# insert into a values(1, 1);
INSERT 0 1
postgres=# insert into a values(2, 1);
INSERT 0 1
postgres=# insert into a values(3, 1);
ERROR:  Found gp_relation_node entry for relation name a, relation Oid 17267, relfilenode 17267, segment file #1 at PTID: (2,83), PSN: 585 when not expected (aocsam.c:718)  (seg0 vagrant.vm:40000 pid=14809) (cdbdisp.c:1326)
postgres=# select * from a;
ERROR:  gp_relation_node (2) has more entries than pg_aocsseg (0) for relation a (aocssegfiles.c:1919)  (seg0 slice1 vagrant.vm:40000 pid=14809) (cdbdisp.c:1326)
```

**Comment by asimrp** [comment]
Hi @Tao-Ma 

Thank you for finding and reporting the issue.  I'm able to reproduce it and looking into the details.  Create table command is not behaving correctly on segments.  E.g. On one segment I see additional row in `pg_aocsseg` and two additional rows in `gp_persistent_relation_node` catalog tables as compared with master and other segments.

```
gptest=# begin;
BEGIN
gptest=# create temp table a with (appendonly=true, orientation=column) on commit delete rows as select 1::int AS c1, 1::int AS c;
SELECT 1
gptest=# select relid::regclass, segrelid::regclass from pg_appendonly;
 relid |         segrelid          
-------+---------------------------
 a     | pg_aoseg.pg_aocsseg_16527
(1 row)
```

On master and segment 1, there are no rows in `pg_aocsseg` but on segment 0, we can see one row:
```
gptest=# select gp_segment_id, segno, tupcount, modcount, state from pg_aoseg.pg_aocsseg_16527;
 gp_segment_id | segno | tupcount | modcount | state 
---------------+-------+----------+----------+-------
(0 rows)

gptest=# select gp_segment_id, segno, tupcount, modcount, state from gp_dist_random('pg_aoseg.pg_aocsseg_16527');
 gp_segment_id | segno | tupcount | modcount | state 
---------------+-------+----------+----------+-------
             0 |     0 |        1 |        1 |     1
(1 row)
```

Similar observation for `gp_persistent_relation_node`:
```
gptest=# select relfilenode from pg_class where relname = 'a';
 relfilenode 
-------------
       16527
(1 row)

gptest=# select gp_segment_id, relfilenode_oid, segment_file_num, persistent_state, mirror_append_only_loss_eof, mirror_append_only_new_eof, persistent_serial_num from gp_persistent_relation_node where relfilenode_oid = 16527;
 gp_segment_id | relfilenode_oid | segment_file_num | persistent_state | mirror_append_only_loss_eof | mirror_append_only_new_eof | persistent_serial_num 
---------------+-----------------+------------------+------------------+-----------------------------+----------------------------+-----------------------
            -1 |           16527 |                0 |                1 |                           0 |                          0 |                   969
(1 row)

gptest=# select gp_segment_id, relfilenode_oid, segment_file_num, persistent_state, mirror_append_only_loss_eof, mirror_append_only_new_eof, persistent_serial_num from gp_dist_random('gp_persistent_relation_node') where relfilenode_oid = 16527;
 gp_segment_id | relfilenode_oid | segment_file_num | persistent_state | mirror_append_only_loss_eof | mirror_append_only_new_eof | persistent_serial_num 
---------------+-----------------+------------------+------------------+-----------------------------+----------------------------+-----------------------
             1 |           16527 |                0 |                1 |                           0 |                          0 |                   973
             0 |           16527 |              128 |                1 |                           0 |                          0 |                   983
             0 |           16527 |                0 |                1 |                           0 |                          0 |                   978
(3 rows)
```

**Comment by asimrp** [comment]
FIxed by PR #2046

**Comment by Tao-Ma** [comment]
Cool, happy to have a faster temp table.
